### PR TITLE
Added an Audit step for FreeBSD and DragonFly packagees.

### PR DESCRIPTION
**Pull request by wolpert**
Allows for auditing the packages to be disabled since they are breaking steps. Current behaivor is the default, where if the audit fails topgrade stops. Can be disabled in the [misc] section independenly from other sections.

Probably we should have a document explaining all the Steps in the system. Maybe as a separate PR?

This is a potential fix for the bug I reported here: https://github.com/topgrade-rs/topgrade/issues/639

## Standards checklist:

- [X] The PR title is descriptive.
- [X] I have read `CONTRIBUTING.md`
- [X] The code compiles (`cargo build`)
- [X] The code passes rustfmt (`cargo fmt`)
- [X] The code passes clippy (`cargo clippy`)
- [X] The code passes tests (`cargo test`)
- [X] *Optional:* I have tested the code myself
 
## For new steps
- [X] *Optional:* Topgrade skips this step where needed
- [ ] *Optional:* The `--dry-run` option works with this step
- [ ] *Optional:* The `--yes` option works with this step if it is supported by 
  the underlying command

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

### CODE REVIEW

**Comment by wolpert** [comment]
BTW, tested in FreeBSD.

With default configuration file, there is no change in behavior.

When the following is added to the configuration file, the Audit step is removed.
```toml
[misc]
disable = ["audit"]
```

**Review by SteveLauC — CHANGES_REQUESTED** [review]
Hi, thanks for this patch!

If you want to make audit a step, you should also call `print_separator()` in that function:

```diff
pub fn audit_packages(ctx: &ExecutionContext) -> Result<()> {
    let sudo = require_option(ctx.sudo().as_ref(), REQUIRE_SUDO.to_string())?;
    
    print_separator("FreeBSD Audit");
    
    Command::new(sudo)
        .args(["/usr/sbin/pkg", "audit", "-Fr"])
        .status_checked()
}
```

File: src/steps/os/dragonfly.rs [hunk 1]
```diff
@@ -19,7 +19,9 @@ pub fn upgrade_packages(ctx: &ExecutionContext) -> Result<()> {
 
 pub fn audit_packages(ctx: &ExecutionContext) -> Result<()> {
     let sudo = require_option(ctx.sudo().as_ref(), REQUIRE_SUDO.to_string())?;
-    println!();
+
+    print_separator("DragonFly BSD Packages");
```
Comment (SteveLauC) [inline src/steps/os/dragonfly.rs:23]:
```suggestion
    print_separator("DragonFly BSD Audit");
```

**Comment by wolpert** [comment]
> If you want to make audit a step, you should also call `print_separator()` in that function:

Good idea. Just updated with the change to both FreeBSD and DragonFly. Thanks!

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**